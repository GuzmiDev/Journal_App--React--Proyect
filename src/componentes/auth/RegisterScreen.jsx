import { Link } from "react-router-dom";
import { useForm } from "../../hooks/useForm";

const RegisterScreen = () => {
  const [formValues, handleInputChange] = useForm({
    name: "",
    email: "",
    password: "",
    password2: "",
  });

  const handleRegister = (e) => {
    e.preventDefault();
    console.log(formValues);
  };

  return (
    <>
      <h3 className="auth__title mb-5">Login</h3>
      <form onSubmit={handleRegister}>
        <input
          className="auth__input"
          type="text"
          placeholder="Name"
          name="name"
          autoComplete="off"
          value={formValues.name}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="text"
          placeholder="Email"
          name="email"
          autoComplete="off"
          value={formValues.email}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Password"
          name="password"
          autoComplete="off"
          value={formValues.password}
          onChange={handleInputChange}
        />
        <input
          className="auth__input"
          type="password"
          placeholder="Confirm"
          name="password2"
          autoComplete="off"
          value={formValues.password2}
          onChange={handleInputChange}
        />
        <button className="btn btn-primary btn-block" type="submit">
          Register
        </button>
        <Link className="link mt-5" to="/auth/login">
          Already regitered?
        </Link>
      </form>
    </>
  );
};

export default RegisterScreen;
