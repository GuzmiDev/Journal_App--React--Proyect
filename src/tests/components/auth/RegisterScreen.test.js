/**
 *@jest-environment node
 */
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import { mount } from "enzyme";
import "jsdom-global/register";
import configureStore from "redux-mock-store"; //ES6 modules
import thunk from "redux-thunk";
import RegisterScreen from "../../../componentes/auth/RegisterScreen";
import { types } from "../../../types/types";

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
  auth: {},
  ui: {
    loading: false,
    msg: null,
  },
};

const store = mockStore(initState);

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <RegisterScreen />
    </MemoryRouter>
  </Provider>
);

describe("Pruebas en <RegisterScreen>", () => {
  beforeEach(() => {
    store.clearActions();
    jest.clearAllMocks();
  });
  test("debe renderizarse correctamente", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("debe de hacer el dispatch de la acciÃ³n respectiva", () => {
    const nameField = wrapper.find("input[name='name']");

    nameField.simulate("change", {
      target: {
        value: "",
        name: "name",
      },
    });

    wrapper.find("form").simulate("submit");

    const actions = store.getActions();
    expect(actions[0]).toEqual({
      type: types.uiSetError,
      payload: "Name is required",
    });
  });
});
