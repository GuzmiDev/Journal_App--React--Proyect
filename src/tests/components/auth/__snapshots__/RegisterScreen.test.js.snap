// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <RegisterScreen> debe renderizarse correctamente 1`] = `
Array [
  <h3
    className="auth__title mb-5"
  >
    Register
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value=""
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm"
      type="password"
      value=""
    />
    <button
      className="btn btn-primary btn-block"
      type="submit"
    >
      Register
    </button>
    <a
      className="link mt-5"
      href="/auth/login"
      onClick={[Function]}
    >
      Already regitered?
    </a>
  </form>,
]
`;
